### PR TITLE
Add PHP 8 testing to GitHub action

### LINKED ISSUE #21: PHP 8 Support
Add allowed failure testing for PHP 8 to GitHub Action
```
    continue-on-error: ${{ matrix.experimental }}
    strategy:
      matrix:
        include:
          - php: '8.0'
            wordpress: 'latest'
            experimental: true
```

### PR DESCRIPTION
**Pull request by grappler**
Fixes https://github.com/wearerequired/digest/issues/21

### CODE REVIEW

File: .github/workflows/phpunit.yml [hunk 1]
```diff
@@ -25,6 +25,7 @@ jobs:
           --health-timeout 5s
           --health-retries 3
 
+    continue-on-error: ${{ matrix.experimental }}
```
Comment (ocean90) [inline .github/workflows/phpunit.yml:28]: To make this work the default matrix needs an `experimental: [false]` added.

**Review by ocean90 — CHANGES_REQUESTED** [review]

**Review by ocean90 — CHANGES_REQUESTED** [review]
@grappler You should verify that the action is actually running. You can see on https://github.com/wearerequired/digest/actions/runs/679358502  that it's not.

There are two configurations left which don't define `experimental`.

**Comment by ocean90** [comment]
Tests for PHP 8.0 are now running but there's a valid error which needs to be fixed:

```
There was 1 error:

1) Required\Digest\Tests\CoreUpdate::test_no_entries
Required parameter $event follows optional parameter $user

/home/runner/work/digest/digest/inc/Message/CoreUpdate.php:32
/home/runner/work/digest/digest/tests/phpunit/tests/CoreUpdate.php:10
```

https://github.com/wearerequired/digest/blob/909f2fd832d7e2612bcb689c1130cc02e7f3d54e/inc/Message/CoreUpdate.php#L23-L36

File: .github/workflows/phpunit.yml [hunk 2]
```diff
@@ -25,16 +25,21 @@ jobs:
...
     strategy:
       fail-fast: false
       matrix:
         php: ['7.2', '7.3', '7.4']
```
Comment (ocean90) [inline .github/workflows/phpunit.yml:32]: Looks like we can now remove the `experimental` flag and add 8.0 to the list by default.

File: .github/workflows/phpunit.yml [hunk 3]
```diff
@@ -77,16 +77,11 @@ jobs:
       - name: Remove coding standards
         run: composer remove --dev wearerequired/coding-standards --ignore-platform-reqs
 
-      - name: Install dependencies (PHP < 8.0)
-        if: ${{ matrix.php < '8.0' }}
+      - name: Install dependencies
```
Comment (ocean90) [inline .github/workflows/phpunit.yml]: This doesn't work, see https://github.com/wearerequired/digest/pull/23/commits/ec23ce1e153995ae8f7037fa788d834085bbc508 and https://github.com/wearerequired/digest/pull/23/commits/34103806c3a774cbb14c0c8428d72b8983c9f90c. Why not keeping it as is?

Comment (grappler) [reply .github/workflows/phpunit.yml]: I was trying to understand the reason why it is needed. I have documented it in https://github.com/wearerequired/digest/pull/23/commits/ad4bcebfa61f20f7375b8b629e8bdd85797308cd

File: .github/workflows/phpunit.yml (hunk 2 above)
Comment (grappler) [reply .github/workflows/phpunit.yml:32]: I would like to keep the `experimental` flag for the moment.

**Pull request merged**